instrumentation/pubsub-2.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/pubsub/PubSubAttributesMapSetter.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.pubsub;
+
+import com.google.pubsub.v1.PubsubMessage;
+import io.opentelemetry.context.propagation.TextMapSetter;
+import java.lang.reflect.Field;
+import java.util.Map;
+
+public enum PubSubAttributesMapSetter implements TextMapSetter<PubsubMessage> {
+  INSTANCE;
+
+  @Override
+  public void set(PubsubMessage carrier, String key, String value) {
+    try {
+      Class cls = carrier.getClass();
```

Comment:
Can we extract this to a utility that gets an instance, a private attribute name, and wraps this neatly?